index-scheduler/src/index_mapper.rs

```diff
@@ -83,7 +83,14 @@ impl IndexMapper {
 
                 let index_path = self.base_path.join(uuid.to_string());
                 fs::create_dir_all(&index_path)?;
-                self.create_or_open_index(&index_path)
+                let index = self.create_or_open_index(&index_path)?;
+
+                // TODO: this is far from perfect. If the caller don't commit or fail his commit
+                // then we end up with an available index that should not exist and is actually
+                // not available in the index_mapping database.
+                self.index_map.write().unwrap().insert(uuid, Available(index.clone()));
```

Comment:
Can't we take the `wtxn` by ownership and commit on our side, only inserting the index in the hashmap if it worked?

Reply:
Also, shouldn't we check that the value is not already in a `BeingDeleted` state, and don't insert it if so?

Reply:
But we actually want to insert it even if it’s in a BeingDeleted state no?
It'll still be deleted later.

> Can't we take the wtxn by ownership and commit on our side, only inserting the index in the hashmap if it worked?

Maybe we could, maybe not, I think it would change a lot of code and I don’t really have the time rn but we can check it together in a few minutes